Dockerfile

```diff
@@ -18,16 +18,19 @@ WORKDIR /opt/nodemcu-firmware
 # - FLOAT_ONLY=1, if you want the floating point firmware
 #
 # What the commands do:
-# - store the Git branch in 'BRANCH'
+# - store the Git branch in 'BRANCH' if IMAGE_NAME not set
 # - unpack esp-open-sdk.tar.gz in a directory that is NOT the bound mount directory (i.e. inside the Docker image)
 # - remove all files in <firmware-dir>/bin
 # - make a float build if !only-integer
 # - copy and rename the mapfile to bin/
 # - make an integer build
 # - copy and rename the mapfile to bin/
-CMD BRANCH="$(git rev-parse --abbrev-ref HEAD)" && \
-    BUILD_DATE="$(date +%Y%m%d-%H%M)" && \
-    if [ -z "$IMAGE_NAME" ]; then IMAGE_NAME=${BRANCH}_${BUILD_DATE}; else true; fi && \
+CMD BUILD_DATE="$(date +%Y%m%d-%H%M)" && \
+    if [ -z "$IMAGE_NAME" ]; then \
```

Comment:
Fair enough, I get it. I'm familiar with Git submodules but I was looking at this from the wrong angle.

Altering the order of the statements does no harm. Hence, I don't mind changing them. Setting `BUILD_DATE` should be inside the if-statement as well, though.